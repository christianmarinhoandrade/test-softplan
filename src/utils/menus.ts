export const getMenu = (
) => {
  return [
    {
      title: 'teste',
      path: '/teste',
      name: 'agendamento'
    }
  ]
}
