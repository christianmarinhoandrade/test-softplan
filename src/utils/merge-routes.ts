import { RouteProps } from 'react-router-dom'
const routesContext = require.context('../pages', true, /.routes./)

export type RouteMergeProps = {
  name?: string
  path?: string
} & RouteProps

const routes: RouteMergeProps[] = []

routesContext.keys().forEach((route) => {
  routes.push(...routesContext(route).default)
})

export default routes
