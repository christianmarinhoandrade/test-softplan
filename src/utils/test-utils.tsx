import { ReactElement } from 'react'
import { render } from '@testing-library/react'

import { Provider as ProviderRedux } from 'react-redux'
import store from 'store'

import ThemeProvider from 'providers/theme'

const customRender = (ui: ReactElement) =>
  render(
    <ProviderRedux store={store}>
      <ThemeProvider>{ui}</ThemeProvider>
    </ProviderRedux>
  )

export * from '@testing-library/react'
export { customRender as render }
