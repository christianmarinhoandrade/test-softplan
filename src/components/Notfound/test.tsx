import 'match-media-mock'
import { render } from 'utils/test-utils'

import NotFoundComponent from '.'

describe('<NotFound />', () => {
  it('should render not found', async () => {
    const { container } = render(<NotFoundComponent />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
