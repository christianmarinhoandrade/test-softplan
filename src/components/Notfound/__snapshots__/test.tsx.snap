// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render not found 1`] = `
.c0 {
  margin-top: 2rem;
}

<div
  class="c0"
>
  <h1
    class="title"
  >
    Erro: 404
  </h1>
  <p>
    Página não encontrada.
  </p>
</div>
`;
